cachito/web/api_v1.py

```diff
@@ -211,15 +212,24 @@ def download_archive(request_id):
         )
         raise InternalServerError()
 
+    checksum = compute_file_checksum(get_hasher(), str(bundle_dir.bundle_archive_file))
+    store_checksum = bundle_dir.bundle_archive_checksum.read_text(encoding="utf-8")
+    if checksum != store_checksum:
+        msg = f"Checksum of bundle archive {bundle_dir.bundle_archive_file} has been changed."
+        flask.current_app.logger.error(msg)
+        raise InternalServerError(msg)
```

Comment:
IMO the error message should not show the full file path, maybe just the file name or not at all (but it would be good to show the full file path in the log message).

And I would suggest to change the wording from `has been changed` to `has changed` - the first one kind of sounds like the checksum was changed intentionally